src/repositories/PostRepository.ts

```diff
@@ -90,6 +91,16 @@ export class PostRepository extends AbstractRepository<PostModel> {
       .getMany();
   }
 
+  public async filterPostsByPrice(lowerBound: number, upperBound: number): Promise<PostModel[]> {
+    return await this.repository
+      .createQueryBuilder("post")
+      .leftJoinAndSelect("post.user", "user")
+      .where("post.price >= :lowerBound", {lowerBound: lowerBound})
+      .andWhere("post.price <= :upperBound", {upperBound: upperBound})
+      .andWhere("post.archive = false")
```

Comment:
should also have `.andWhere("post.user != user.id)` to eliminate people seeing their own posts.